autoextract_spiders/middlewares.py

```diff
@@ -0,0 +1,91 @@
+"""
+Writes a single json file for each issued item in the target s3://bucket/folder.
+
+Usage:
+
+SPIDER_MIDDLEWARES = {
+    'autoextract_spiders.middlewares.S3ItemWriteMiddleware': 1,
```

Comment:
What's the reason for using custom middleware, and not  built-in Scrapy s3 export features? Is it because of specific output structure, or something else?

Reply:
I've replied in the main thread, thanks!